token/provider.go

```diff
@@ -26,7 +27,7 @@ import (
 // (1) a nonempty token and a nil error, or
 //
 // (2) an empty string and a non-nil error.
-type Provider func(context.Context) (string, error)
+type Provider = httpclient.TokenProvider
```

Comment:
Just noticed this now, but does this break/change public API? If the signatures resolve to the same thing it's probably fine in most call sites, but this may technically change the declared type.

Reply:
Yeah they are the same underlying type. This change avoids a bunch of product code that gets a `token.Provider` and has to cast it to `httpclient.TokenProvider` to actually use it. This change avoids the cast by making them the same type.